src/applications/vaos/express-care/redux/actions.js

```diff
@@ -145,14 +145,35 @@ export function routeToPageInFlow(flow, history, current, action) {
       type: FORM_PAGE_CHANGE_STARTED,
     });
 
-    const nextAction = flow[current][action];
+    const state = getState();
     let nextPage;
 
-    if (typeof nextAction === 'string') {
-      nextPage = flow[nextAction];
+    if (action === 'next') {
+      const nextAction = flow[current][action];
+
+      if (typeof nextAction === 'string') {
+        nextPage = flow[nextAction];
+      } else {
+        const nextStateKey = await nextAction(getState(), dispatch);
+        if (
```

Comment:
I think you'll need to do the previousPages logic whether the nextAction is a string or a function.

Reply:
ah good catch

Reply:
This is outside of the string or function check already